src/main/kotlin/no/nav/tiltaksarrangor/controller/request/EndringsmeldingRequest.kt

```diff
@@ -0,0 +1,64 @@
+package no.nav.tiltaksarrangor.controller.request
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import no.nav.tiltaksarrangor.model.DeltakerStatusAarsak
+import java.time.LocalDate
+
+data class EndringsmeldingRequest(
+	@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+	@JsonSubTypes(
+		JsonSubTypes.Type(value = Innhold.LeggTilOppstartsdatoInnhold::class, name = "LEGG_TIL_OPPSTARTSDATO"),
+		JsonSubTypes.Type(value = Innhold.EndreOppstartsdatoInnhold::class, name = "ENDRE_OPPSTARTSDATO"),
+		JsonSubTypes.Type(value = Innhold.ForlengDeltakelseInnhold::class, name = "FORLENG_DELTAKELSE"),
+		JsonSubTypes.Type(value = Innhold.AvsluttDeltakelseInnhold::class, name = "AVSLUTT_DELTAKELSE"),
+		JsonSubTypes.Type(value = Innhold.DeltakerIkkeAktuellInnhold::class, name = "DELTAKER_IKKE_AKTUELL"),
+		JsonSubTypes.Type(value = Innhold.EndreDeltakelseProsentInnhold::class, name = "ENDRE_DELTAKELSE_PROSENT")
+	)
+	val innhold: Innhold
+) {
+	enum class EndringsmeldingType {
+		LEGG_TIL_OPPSTARTSDATO,
+		ENDRE_OPPSTARTSDATO,
+		FORLENG_DELTAKELSE,
+		AVSLUTT_DELTAKELSE,
+		DELTAKER_IKKE_AKTUELL,
+		ENDRE_DELTAKELSE_PROSENT
+	}
+
+	sealed class Innhold {
+		data class LeggTilOppstartsdatoInnhold(
+			val type: EndringsmeldingType = EndringsmeldingType.LEGG_TIL_OPPSTARTSDATO,
+			val oppstartsdato: LocalDate
+		) : Innhold()
+
+		data class EndreOppstartsdatoInnhold(
+			val type: EndringsmeldingType = EndringsmeldingType.ENDRE_OPPSTARTSDATO,
+			val oppstartsdato: LocalDate
+		) : Innhold()
+
+		data class EndreDeltakelseProsentInnhold(
+			val type: EndringsmeldingType = EndringsmeldingType.ENDRE_DELTAKELSE_PROSENT,
+			val deltakelseProsent: Int,
+			val gyldigFraDato: LocalDate?
+		) : Innhold()
+
+		data class ForlengDeltakelseInnhold(
+			val type: EndringsmeldingType = EndringsmeldingType.FORLENG_DELTAKELSE,
+			val sluttdato: LocalDate
+		) : Innhold()
+
+		data class AvsluttDeltakelseInnhold(
+			val type: EndringsmeldingType = EndringsmeldingType.AVSLUTT_DELTAKELSE,
+			val sluttdato: LocalDate,
+			val aarsak: DeltakerStatusAarsak.Type,
+			val beskrivelse: String?
```

Comment:
Forstår ikke helt hvorfor beskrivelse er i innholdet, det er jo en beskrivelse av årsaken ANNET sånn rent teknisk sett.

Reply:
Jeg har for det meste fulgt api-definisjonene fra Magnus her: https://mevv.notion.site/amt-tiltaksarrangor-api-02009867f0d646edb237613c7bcbb11c#5a4049791c724b24a632071466687f3f

Vi kan godt ta en prat om det kanskje burde vært annerledes..?

Reply:
Jeg er litt enig med deg når jeg ser hvordan jeg modellerte endringsmeldingene man får av get-endepunktet.. Ingen grunn til å pakke ut årsaken sånn som jeg gjør her da egentlig. Endrer :)